test/pkg.jl

```diff
@@ -100,6 +100,8 @@ temp_pkg_dir() do
     str = chomp(takebuf_string(iob))
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
+    # Checking to see dependency reading
+    @test Pkg.dependencies("Example") == Dict("julia" => Pkg.Types.VersionSet(v"0.3.0", typemax(VersionNumber)))
```

Comment:
What if the default behavior of `dependencies` wasn't to check the latest version, but the installed version? I'm thinking something like:

```julia
function dependencies(packagename::AbstractString)
    cd(Pkg.dir()) do
        if Pkg.Read.isinstalled(packagename)
            reqpath = joinpath(Pkg.dir(packagename), "REQUIRE")
            if !isfile(reqpath)
                throw(PkgError("No file found for installed package $packagename at $reqpath."))
            end
            Pkg.Reqs.parse(readlines(reqpath))
        else
            Pkg.Read.latest()[packagename].requires
        end
    end
end
function dependencies(packagename::AbstractString, version::VersionNumber)
    cd(Pkg.dir()) do
        Pkg.Read.available(packagename)[version].requires
    end
end
```

I think this makes more sense as default behavior, and allows for flexibility through the second method def. 

In terms of the test, it should be like pinning the test to whichever version of Example.jl would be installed on the version of Julia being tested. Then the test will only need updating if/when repository structure changes and the Pkg module will need to be updated anyways.